src/globus_sdk/services/auth/id_token_decoder.py

```diff
@@ -0,0 +1,173 @@
+
+        jwt_audience = audience if audience is not None else self.default_audience
+        jwt_params = jwt_params or {}
+
+        openid_configuration = self.get_openid_configuration()
+        jwk = self.get_jwk()
+
+        signing_algos = openid_configuration["id_token_signing_alg_values_supported"]
+
+        return jwt.decode(
+            id_token,
+            key=jwk,
+            algorithms=signing_algos,
+            audience=jwt_audience,
+            options=jwt_params,
+            leeway=leeway,
+        )
+
+    @property
+    def default_audience(self) -> str | None:
+        """The default 'audience' for verification of the JWT 'aud' claim."""
+        return None
+
+    @abc.abstractmethod
+    def get_openid_configuration(self) -> dict[str, t.Any]:
+        """Fetch the OpenID Configuration for Globus Auth."""
+
+    @abc.abstractmethod
+    def get_jwk(self) -> RSAPublicKey:
+        """Fetch the JWK for Globus Auth."""
+
+
+class DefaultIDTokenDecoder(IDTokenDecoder):
+    """
+    The default implementation of ID token decoding.
+
+    The default decoder implementation accepts a client on init, and will reach
+    out to Globus Auth to fetch OIDC configuration data dynamically, as needed.
+    It also has methods to store these values, to avoid the HTTP callouts at time
+    of use.
+
+    The main method of a decoder is ``decode()``, which accepts an ``id_token`` string
+    and parses it into a dict.
+
+    :param auth_client: The client which should be used to callout to Globus Auth as
+        needed. Any AuthClient or AuthLoginClient will work for this purpose.
+    """
+
+    def __init__(self, auth_client: SupportsJWKMethods) -> None:
+        self._auth_client = auth_client
+        self._openid_configuration: dict[str, t.Any] | None = None
+        self._jwk: RSAPublicKey | None = None
+
+    def store_openid_configuration(
+        self, openid_configuration: dict[str, t.Any] | GlobusHTTPResponse
+    ) -> None:
+        """
+        Store openid_configuration data for future use in ``decode()``.
+
+        :param openid_configuration: The configuration data, as fetched via
+            :meth:`AuthClient.get_openid_configuration`
+        """
+        if isinstance(openid_configuration, GlobusHTTPResponse):
+            self._openid_configuration = openid_configuration.data
+        else:
+            self._openid_configuration = openid_configuration
+
+    def store_jwk(self, jwk: RSAPublicKey) -> None:
+        """
+        Store a JWK for future use in ``decode()``.
+
+        :param jwk: The JWK, as fetched via :meth:`AuthClient.get_jwk`
+            with ``as_pem=True``.
+        """
+        self._jwk = jwk
+
+    @property
+    def default_audience(self) -> str | None:
+        """
+        The audience for JWT verification defaults to the client's client ID.
+        """
+        if not hasattr(self._auth_client, "client_id"):
+            return None
+        return self._auth_client.client_id
```

Comment:
The protocol mandates the client_id attribute exist ([ref](https://github.com/globus/globus-sdk-python/blob/main/src/globus_sdk/services/auth/_common.py#L90)); it just might be None (unless I don't understand protocols which is very possible).

```suggestion
        return self._auth_client.client_id
```